oaknational/oak-ai-lesson-assistant
# feat: react scan

## Description

- Introduced react-scan for performance monitoring
- Added useReactScan hooks that can be dropped into component to track render of all or passed in component. Will console table of all components or log of individual component.

- Integrated an example Playwright tests to validate unnecessary renders. (unfortunately only works locally , so have removed for now)


`dev:react-scan`: Enables React Scan for local performance monitoring.
`dev:react-scan-monitor`: Tracks performance in the React Scan dashboard.
For dashboard access, ask me for login credentials.
React Scan dashboard URL: [React Scan Dashboard](https://dashboard.react-scan.com/project/VsUTvxwo-sMiPvQ6qU5rQ)

https://github.com/aidenybai/react-scan
[![image](https://github.com/user-attachments/assets/00d78a59-bd30-4a94-ac11-f1be260bc322)](https://raw.githubusercontent.com/aidenybai/react-scan/refs/heads/main/.github/assets/demo.gif)

## How to test

Run `dev:react-scan` and go to lesson chat
Run `dev:react-scan-monitor` and interact with the site, go to [React Scan Dashboard](https://dashboard.react-scan.com/project/VsUTvxwo-sMiPvQ6qU5rQ) to see render report

## Screenshots

react-scan monitoring dashboard

<img width="1464" alt="image" src="https://github.com/user-attachments/assets/72e2c646-ded8-4ac2-8de8-175f8f4c076a" />

logs from useReactScan

<img width="824" alt="image" src="https://github.com/user-attachments/assets/b923154a-0c36-4f69-8083-f7dae065f4d4" />

react-scan overlay with number of renders

<img width="824" alt="image" src="https://github.com/user-attachments/assets/112edff3-8a79-4abf-8e40-7739b4023733" />


## Checklist
